packages/dataviews/src/dataviews-layouts/table/use-is-scrolled-end.ts

```diff
@@ -0,0 +1,56 @@
+/**
+ * WordPress dependencies
+ */
+import { useEffect, useState } from '@wordpress/element';
+import { isRTL } from '@wordpress/i18n';
+
+const isScrolledToEnd = ( element: Element ) => {
+	if ( isRTL() ) {
+		const scrollLeft = Math.abs( element.scrollLeft );
+		return scrollLeft <= 1;
+	}
+
+	return element.scrollLeft + element.clientWidth >= element.scrollWidth - 1;
+};
+
+export function useIsScrolledEnd( {
+	selector,
+	enabled = false,
+}: {
+	selector: string;
+	enabled?: boolean;
+} ): boolean {
+	const [ isScrolledEnd, setIsScrolledEnd ] = useState( false );
+
+	useEffect( () => {
+		if ( typeof window === 'undefined' || ! enabled ) {
+			return () => {};
+		}
+
+		const scrollContainer = document.querySelector( selector );
+
+		const handleIsScrolledEnd = () => {
+			if ( scrollContainer ) {
+				setIsScrolledEnd( isScrolledToEnd( scrollContainer ) );
```

Comment:
Should this be throttled?

Reply:
Wrapped the `handleIsScrolledEnd` function with the `useDebounce`.